weatherApp/src/screens/CityScreen/index.tsx

```diff
@@ -0,0 +1,106 @@
+import React, {useEffect, useState} from 'react';
+import styles from './styles';
+import {SafeAreaView, Text} from 'react-native';
+import Header from '../../components/Header';
+import DataSection from '../../components/DataSection';
+import {City} from '../../stores/CityStore';
+import {Weather} from '../../stores/WeatherStore';
+import {WeatherForecastResponse, weatherApi} from '../../services/WeatherAPI';
+import {
+  useCityStore,
+  useLanguageStore,
+  useWeatherStore,
+} from '../../contexts/StoreContext';
+import {observer} from 'mobx-react';
+
+type CityScreenProps = {
+  city?: City;
+};
+
+const CityScreen = (props: CityScreenProps) => {
+  const cityStore = useCityStore();
+  const weatherStore = useWeatherStore();
+  const languageStore = useLanguageStore();
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string>('');
+  const [currentWeather, setCurrentWeather] = useState<Weather>();
```

Comment:
logic should be in a store or hooks. The screen is responsible for showing content.